### PR TITLE
Scroll area command

### PR DESCRIPTION
**Pull request by leekevinyg**
A command to help a user scroll a particular area on a page, when multiple scrollable areas are available. 

Example use case: Scroll a Spotify playlist when both the "Playlists" area on the left, as well as the songs in the actual playlist are scrollable.

### CODE REVIEW

File: src/scroll-area.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
+export default {
```
Comment (sljavi) [inline src/scroll-area.js]: you now here can import the tools in a regular way
e.g.
```
import jQuery from 'jquery'
import domElementTypes from 'dom-element-types'
```

File: src/scroll-area.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        });
+
+        for (let i = 0; i < scrollableElements.length; i++) {
+          const scrollableElement = scrollableElements[i];
+          if (!scrollableElement.classList.contains('hands-free-scrollable')) {
+            scrollableElement.className += ' hands-free-scrollable';
+          }
+        }
+
+        if (scrollableElements.length === 0) {
```
[the remark this answers is not in the document]
Comment (leekevinyg) [reply src/scroll-area.js]: Ahhh good idea! I assume you mean when ```scrollableElements.length === 1``` right? I have added the following to cover that case, let me know if this matches what you meant.

```
        if (scrollableElements.length === 1) {
          const isBody = jQuery(scrollableElements[0]).is('body');
          if (isBody) {
            return {
              context: 'scroll-area',
              selectedElement: jQuery('html', 'body'),
            };
          }
          return {
            context: 'scroll-area',
            selectedElement: scrollableElements[0],
          };
        }
```

File: src/scroll-area.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
+export default {
+  name: 'i18n-name',
+  description: 'i18n-description',
```
[the remark this answers is not in the document]
Comment (leekevinyg) [reply src/scroll-area.js]: Nice! I added the same one as scroll.js as they are similar actions.

File: src/scroll-area.js [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+        const scrollableElements = allElementsOnPage.filter((element) => {
+          const computedStyle = getComputedStyle(element);
+          const canScrollNow = element.scrollHeight > element.clientHeight
+              && (computedStyle.overflowY === 'auto' || computedStyle.overflowY === 'scroll');
+          return canScrollNow;
+        });
+
+        for (let i = 0; i < scrollableElements.length; i++) {
+          const scrollableElement = scrollableElements[i];
+          if (!scrollableElement.classList.contains('hands-free-scrollable')) {
```
[the remark this answers is not in the document]
Comment (leekevinyg) [reply src/scroll-area.js]: Great idea. Fixed.

File: src/scroll-area.js [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+      action: ({ tools }) => {
+        const allElementsOnPage = Array.from(document.getElementsByTagName('*'));
+        const scrollableElements = allElementsOnPage.filter((element) => {
+          const computedStyle = getComputedStyle(element);
+          const canScrollNow = element.scrollHeight > element.clientHeight
+              && (computedStyle.overflowY === 'auto' || computedStyle.overflowY === 'scroll');
+          return canScrollNow;
+        });
+
+        for (let i = 0; i < scrollableElements.length; i++) {
```
[the remark this answers is not in the document]
Comment (leekevinyg) [reply src/scroll-area.js]: Fixed.

**Comment by sljavi** [comment]
@leekevinyg awesome! merging it!

**Comment by leekevinyg** [comment]
Thanks @sljavi ! Please let me know when the command is released on the chrome store  :)

**Comment by sljavi** [comment]
@all-contributors please add @leekevinyg for implementing the voice command "scroll area"

**Comment by sljavi** [comment]
@all-contributors please add @leekevinyg for code a4fc51ea658f2c5904cc52c17a4f5a02e017464d 7e7441aa698922d834873bda8f339bca4ea8fca0 dff23bbeac4f503fbc07391b78e522d6ba24aff2

**Comment by sljavi** [comment]
@all-contributors please add @iMatheusouza for translations

**Pull request merged**